pipelines/src/pipelines/utils/upload_pipeline.py

```diff
@@ -12,21 +12,28 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import argparse
+from typing import List
 from kfp.registry import RegistryClient
 
-if __name__ == "__main__":
+
+def upload_pipeline(args: List[str] = None):
```

Comment:
let's move arg parsing to the L39 so that 
```suggestion
def upload_pipeline(host:str, file_name:str, tags:List[str]):
```

Reply:
as unit tests go, you can do

```python

def main():
    parser = argparse.ArgumentParser()
    # ...

if __name__ == "__main__":
    main()
```

to test arg parsing separately